### DISCUSSION TITLE
DMX learning

### DISCUSSION THREAD

**Discussion by Mat-Moo** [forkineye/ESPixelStick#608] [General]
```FW Version
4.x-dev Build 
Date Dec 12 2022 - 09:32:31
Architecture ESP8266Board 
Named 1_miniFlash 
Size 4194304
Flash ID 1a8534
```
I'm just playing around and trying to understand DMX on EspixelStick 4 firmware. I've got an old disco light I'm trying to play with, so bumped latest dev on there, setup my input as e1.31, universe 1 first channel 1 510 per universe, DMX out channels 64 and port 2 (not got anything wired up yet). This is on a dedicated fixed ip address, and then I'm using da_E131 Test Source to send data, unicast same IP same port, but in diagnostics I see nothing? Should that be showing me some data?

**Comment by MartinMueller2003** [comment]
Look at the status page. Is the packet count going up? If not, then nothing is making it to the ESP.

**Reply by Mat-Moo** [reply]
hmmm good shout... nothing... odd
![image](https://user-images.githubusercontent.com/908888/209398442-424151e2-bb1d-4aae-bc61-c02ccb8cb24c.png)

**Comment by MartinMueller2003** [comment]
Those zeros on the packet counts indicate there are no E1.31 packets arriving at the ESP.

**Reply by Mat-Moo** [reply]
Yea, it's odd I restarted the app and seeing it as expected - now I need a dmx cable to test with!

**Comment by Mat-Moo** [comment]
So got the DMX stuff all working as expected, now challenge 2 was to try and transmit on RF to talk to a random led bracelet I had (Just for kicks). So I bought a transmitter, added on and nothing, hmm, realised I had gone 2.4ghz and not 433mhz so got another, hmmm still nothing. Couple of things I'm thinking #1  baud rate is too high, these cheap transmitters work at very slow speeds, but even if I try serial output the interface doesn't allow below 32k baud rate. #2 my transmitter probably doesnt need to go through a ttl->rs484 convertor as it only has one input so I'm thinking I can just transmit serial (Unless dmx protocol adds other stuff?) - Just brain dump, thoughts welcome..

**Comment by MartinMueller2003** [comment]
DMX protocol is fixed at 250,000 bps, has specific timing to indicate start of frame and has a max frame size of 512 intensity values. It looks like you want to use something like Renard or generic serial.
